src/apps/workspaces/crud/user_applet_access.py

```diff
@@ -372,6 +374,7 @@ async def upsert_user_applet_access(self, schema: UserAppletAccessSchema):
                 "updated_at": datetime.utcnow(),
                 "meta": stmt.excluded.meta,
             },
+            where=where,
         )
```

Comment:
need to add "returning" statement to identify if the record was inserted or not. In not, raise UniqueConstraintViolation error which should be catched in service and muted

Reply:
It updated.